### PR TITLE
Adding support for Home Assistant

### PR DESCRIPTION
**Pull request by emilywotruba**
## Description
Created new custom service for Home Assistant and added documentation.
Allows displaying the name and version of the instance, as well as the count of entities:

screenshot

Fixes # N/A

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I've read & comply with the [contributing guidelines](https://github.com/bastienwirtz/homer/blob/main/CONTRIBUTING.md)
- [x] I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers. 
- [x] I have made corresponding changes to the documentation (README.md).
- [x] I've checked my modifications for any breaking changes, especially in the `config.yml` file

### CODE REVIEW

File: src/components/services/HomeAssistant.vue [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      if (!this.item.separator) this.item.separator = " ";
+      if (!this.item.logo) this.item.logo = `${this.item.url}/static/icons/favicon-192x192.png`;
+      this.fetchServerStats();
+    }
+  },
+  methods: {
+    fetchServerStatus: async function () {
+      const headers = this.headers;
+
+      this.fetch("/api/", { headers })
```
Comment (bastienwirtz) [inline src/components/services/HomeAssistant.vue:96]: You could return the fetch call to make possible to await this.
```suggestion
      return this.fetch("/api/", { headers })
```

File: src/components/services/HomeAssistant.vue [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        }
+      }
+      
+      return details.join(separator);
+    },
+  },
+  created() {
+    this.fetchServerStatus();
+
+    if (!this.item.subtitle && this.status !== "dead") {
```
Comment (bastienwirtz) [inline src/components/services/HomeAssistant.vue:85]: fetchServerStatus is async, and this.status will not have the server status yet. You either have to await or use the promise (.then(....)) to make it reliable.

**Review by bastienwirtz — CHANGES_REQUESTED** [review]
Thanks for your contribution @emilywotruba!
A small issue to fix on the data fetching and we should be good to go.

File: src/components/services/HomeAssistant.vue (hunk 2 above)
Comment (emilywotruba) [reply src/components/services/HomeAssistant.vue]: You're right, i used `then` to resolve this.

**Comment by emilywotruba** [comment]
Thank you for reviewing my request @bastienwirtz. Are your concerns addressed by my new commits?

File: src/components/services/HomeAssistant.vue [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      
+      return details.join(separator);
+    },
+  },
+  created() {
+    this.fetchServerStatus().then(() => {
+      if (!this.item.subtitle && this.status !== "dead") {
+        if (!this.item.items) this.item.items = ["name", "version"];
+        if (!this.item.separator) this.item.separator = " ";
+        if (!this.item.logo) this.item.logo = `${this.item.url}/static/icons/favicon-192x192.png`;
```
Comment (bastienwirtz) [inline src/components/services/HomeAssistant.vue:87]: ESlint is not happy with the prop mutation (cf: https://github.com/bastienwirtz/homer/actions/runs/9179879689/job/25243741529?pr=783#step:6:12), I forgot about that in my first review, sorry. Could you see to use local variable or data variables for example? thanks!

**Review by bastienwirtz — COMMENTED** [review]
Aside from the linting issue, everything looks good @emilywotruba, thanks :pray:

File: src/components/services/HomeAssistant.vue (hunk 3 above)
Comment (emilywotruba) [reply src/components/services/HomeAssistant.vue]: Ah sad, I really liked the idea of automatically grabbing the icon of home assistant if no logo was provided. I will fix it, thank you for your patience.

Comment (bastienwirtz) [reply src/components/services/HomeAssistant.vue]: Yeah that was a good idea! (could be applied to other service too. Thinking about it, in this case it should be harmless, we could disable the lint rules :thinking: 
I'll look into it, we can merge it like that for now, and change it back in the future if we are ok with the prop mutation.

**Review by bastienwirtz — APPROVED** [review]

**Pull request merged**